src/lib/modify-template.ts

```diff
@@ -23,40 +26,57 @@ export function removeNodesFromTemplate(template: Template, nodesToRemove: Set<N
   const {element: {content}, parts} = template;
   const walker = document.createTreeWalker(
     content,
-    NodeFilter.SHOW_ELEMENT | NodeFilter.SHOW_COMMENT |
-        NodeFilter.SHOW_TEXT,
+    walkerNodeFilter,
     null as any,
     false);
   let partIndex = 0;
   let part = parts[0];
   let nodeIndex = -1;
   let removeCount = 0;
   const nodesToRemoveInTemplate = [];
-  // TODO(sorvell): fix type
-  const removalStack: any[] = [];
+  let currentRemovingNode: Node|null = null;
   while (walker.nextNode()) {
     nodeIndex++;
     const node = walker.currentNode as Element;
-    // maybe pop the stack
-    if (walker.previousSibling === removalStack[removalStack.length - 1]) {
-      removalStack.pop();
+    // end removal if stepped past the removing node
+    if (node.previousSibling === currentRemovingNode) {
+      currentRemovingNode = null;
```

Comment:
I think this needs to be the next node in the stack, in the case of nested removal.

Reply:
Added a little comment description at the top of that function. Don't think we care if it's nested since we only need to know when we're ending the removal state of a non-nested node:
* any part inside the removing node gets deactivated
* any part outside a non-nested removing node has its index adjusted

Reply:
Ah, that's right.